pkg/cmd/attestation/verification/sigstore.go

```diff
@@ -65,13 +68,57 @@ func (v *LiveSigstoreVerifier) chooseVerifier(b *bundle.ProtobufBundle) (*verify
 	}
 	issuer := leafCert.Issuer.Organization[0]
 
-	// if user provided a custom trusted root file path, use the custom verifier
-	if v.config.CustomTrustedRoot != "" {
-		customVerifier, err := newCustomVerifier(v.config.CustomTrustedRoot)
+	if v.config.TrustedRoot != "" {
+		customTrustRoots, err := os.ReadFile(v.config.TrustedRoot)
 		if err != nil {
-			return nil, "", fmt.Errorf("failed to create custom verifier: %v", err)
+			return nil, "", fmt.Errorf("unable to read file %s: %v", v.config.TrustedRoot, err)
+		}
+
+		scanner := bufio.NewScanner(bytes.NewReader(customTrustRoots))
+		for scanner.Scan() {
+			// Load each trusted root
+			trustedRoot, err := root.NewTrustedRootFromJSON(scanner.Bytes())
+			if err != nil {
+				return nil, "", fmt.Errorf("failed to create custom verifier: %v", err)
+			}
+
+			// Compare bundle leafCert issuer with trusted root cert authority
+			certAuthorities := trustedRoot.FulcioCertificateAuthorities()
+			for _, certAuthority := range certAuthorities {
+				certAuthRoot := certAuthority.Root
+
+				if len(certAuthRoot.Issuer.Organization) == 0 {
+					continue
+				}
+
+				if certAuthRoot.Issuer.Organization[0] == issuer {
+					// We have our trusted root, but now we need to determine the policy
+					if issuer == PublicGoodIssuerOrg {
+						if v.config.NoPublicGood {
+							return nil, "", fmt.Errorf("Detected public good instance but requested verification without public good instance")
+						}
+						verifier, err := newPublicGoodVerifierWithTrustedRoot(trustedRoot)
```

Comment:
In this case it's really important that we use the trustedRoot provided and *not* the public good one we fetch via TUF.

Reply:
I updated the comment to make this more explicit.